File: LEVEL2/구명보트.java
import java.util.*;
import java.io.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        Arrays.sort(people);
        int min = 0;
        
        for(int idx = people.length - 1; idx >= min; idx--) {
            if(people[idx] + people[min] <= limit) {
                min++;
            }
            
            answer++;
        }
        
        return answer;
    }
}

File: LEVEL2/구명보트.py
def solution(people, limit):
    answer = 0
    people.sort()
    idx = 0
    k = len(people) - 1
    
    while idx <= k:
        answer += 1
        
        if people[idx] + people[k] <= limit:
            idx += 1
        
        k -= 1
        
    return answer